dominode-bootstrapper/dominode_bootstrapper/geonodeadmin.py

```diff
@@ -175,20 +175,58 @@ def from_geonode_manager(cls, geonode_manager: GeoNodeManager):
         )
 
     def list_workspaces(self):
-        raise NotImplementedError  # TODO: Provide implementation
-
-    def create_workspace(self):
-        raise NotImplementedError  # TODO: Provide implementation
+        response = self.client.get(
+            f'{self.base_url}/rest/workspaces',
+            auth=(DEFAULT_USERNAME, DEFAULT_PASSWORD)
+        )
+        return response.json()
 
-    def create_postgis_store(self):
-        raise NotImplementedError  # TODO: Provide implementation
+    def create_workspace(self, name):
+        response = self.client.post(
+            f'{self.base_url}/rest/workspaces',
+            auth=(DEFAULT_USERNAME, DEFAULT_PASSWORD),
+            json={"name": name}
+        )
+        return response.json()
+    def get_workspace(self, name):
+        response = self.client.post(
+            f'{self.base_url}/rest/workspaces/{name}',
+            auth=(DEFAULT_USERNAME, DEFAULT_PASSWORD)
+        )
+        if response.status_code != 200:
+            return None
+        return response.json()
+
+    def create_postgis_store(self,
+                             workspace_name,
+                             store_name,
+                             host,
+                             port,
+                             database,
+                             user,
+                             password):
```

Comment:
- Please follow similar formatting as the rest of the file.
- Please also try to provide type hints for function parameters (ex: `workspace_name: str`) and for function return types, when applicable